### PR TITLE
EZP-28990 - Behat Covering of Object States

### PR DESCRIPTION
**Pull request by m-tyrala**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-28990](https://jira.ez.no/browse/EZP-28990)
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

This PR contains tests that covers Object States in Admin panel.

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

File: src/lib/Behat/BusinessContext/ObjectStatesContext.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * @Then there's :objectStateName on :objectStateGroupName Object States list
+     */
+    public function verfyObjectStateIsOnList(string $objectStateName, string $objectStateGroupName): void
+    {
+        $objectStateGroupPage = PageObjectFactory::createPage($this->utilityContext, ObjectStateGroupPage::PAGE_NAME, $objectStateGroupName);
+        $objectStateGroupPage->verifyIsLoaded();
+        $objectStateExists = $objectStateGroupPage->adminLists['Object States']->table->isElementInTable($objectStateName);
+
+        if (!$objectStateExists) {
+            Assert::fail(sprintf('Element "%s" is not on the "%s" object states list.', $objectStateName, $objectStateGroupName));
```
Comment (mnocon) [inline src/lib/Behat/BusinessContext/ObjectStatesContext.php]: Why not `Assert:assertTrue(isElementInTable,...)`?

File: src/lib/Behat/BusinessContext/ObjectStatesContext.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    /**
+     * @Then there's no :objectStateName on :objectStateGroupName Object States list
+     */
+    public function verfyObjectStateIsNotOnList(string $objectStateName, string $objectStateGroupName): void
+    {
+        $objectStateGroupPage = PageObjectFactory::createPage($this->utilityContext, ObjectStateGroupPage::PAGE_NAME, $objectStateGroupName);
+        $objectStateGroupPage->verifyIsLoaded();
+        try {
+            $objectStateGroupPage->verifyListIsEmpty();
```
Comment (mnocon) [inline src/lib/Behat/BusinessContext/ObjectStatesContext.php]: How about adding a new method: `isListEmpty` (or getItemCount, it's up to you), that will be used by verifyListIsEmpty and here? Catching the Exception seems weird.

File: src/lib/Behat/BusinessContext/ObjectStatesContext.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        } catch (AssertionFailedError $e) {
+            $objectStateExists = $objectStateGroupPage->adminLists['Object States']->table->isElementInTable($objectStateName);
+
+            if ($objectStateExists) {
```
Comment (mnocon) [inline src/lib/Behat/BusinessContext/ObjectStatesContext.php]: Same thing as above (`assertFalse` can be used)

File: src/lib/Behat/PageObject/ObjectStateGroupPage.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+use PHPUnit\Framework\Assert;
+
+class ObjectStateGroupPage extends Page
+{
+    /** @var string Name by which Page is recognised */
+    public const PAGE_NAME = 'Object State Group';
+    /** @var string Name of actual group */
+    public $objectStateGroupName;
+
+    /** @var string locator for container of Content list */
```
Comment (mnocon) [inline src/lib/Behat/PageObject/ObjectStateGroupPage.php]: Please update doc

**Comment by m-tyrala** [comment]
@lserwatka this one can be merged

**Comment by lserwatka** [comment]
@m-tyrala done, you can merge this up

**Pull request merged**